File: WPF/NonUI/NonUIElement.vb
Namespace Common.NonUI

    Public MustInherit Class NonUIElement
        Inherits FrameworkElement

        Private Shared ReadOnly Collapsed As Object = Visibility.Collapsed

        Shared Sub New()
            VisibilityProperty.OverrideMetadata(GetType(NonUIElement), New FrameworkPropertyMetadata(Collapsed, Nothing, Function(D, B) Collapsed))
        End Sub

        Protected Overrides Function MeasureOverride(availableSize As Size) As Size
            Return New Size()
        End Function

    End Class

End Namespace

File: WPF/NonUI/NonUIElement.cs
using System.Windows;

namespace Ks.Common.NonUI
{
    public abstract class NonUIElement : FrameworkElement
    {
        private static readonly object Collapsed = Visibility.Collapsed;

        static NonUIElement()
        {
            VisibilityProperty.OverrideMetadata(typeof(NonUIElement), new FrameworkPropertyMetadata(Collapsed, null, (D, B) => Collapsed));
        }

        protected override Size MeasureOverride(Size availableSize)
        {
            return new Size();
        }
    }
}
